### PR TITLE
CKAN 2.10 + Cypress 12.10.0

### PR DESCRIPTION
**Pull request by nickumia-reisys**
Related to
- https://github.com/GSA/data.gov/issues/4209
- https://github.com/GSA/data.gov/issues/4370
- https://github.com/GSA/data.gov/issues/4369
- https://github.com/GSA/data.gov/issues/4285

Changes:
- `IRoutes` is deprecated and it wasn't doing anything for us since `CKAN 2.9.5`
- Changed references to deprecated `url` to `url_for`
- (BREAKING) `h.flash.pop_messages` is deprecated and replaced with `h.get_flashed_messages`
- (BREAKING) Update references to routes from pylons `organization_read` to new way of `organization.read` ... et cetera.
- Update dataset creation method to return the ID of the new dataset.
- `h.new_activities` is deprecated, so remove code.
- Migrate to cypress `12.10.0` (EACH DEVELOPER NEEDS TO UPDATE THEIR LOCAL SETUP)
- Update cypress tests due to `CKAN 2.10` dataset name inconsistency
- Add (missing?) styles to `datagovtheme.css`
- Deleted custom `resource_preview_custom` function.
  - `h.resource_preview` was mysteriously deleted in CKAN 2.10
- Fix facet list on organization page
- Update `icon-???` to `fa-???`
- Fix `collection_resources` inheritance between `datagovtheme` and `geodatagov` (not sure why it broke in CKAN 2.10)
- Fix flash message styling
- New template override [`templates/snippets/search_form.html`](https://github.com/GSA/ckanext-datagovtheme/pull/162/files#diff-6dcff83c81b9128f224208cc4741ac278baeb6fd8228a3255c6e6b12cdd43984)
- Add new cypress test for contact information on dataset page
- Track upstream `ckanext-spatial` for extension testing

### CODE REVIEW

**Comment by nickumia-reisys** [comment]
@FuhuXia I need help.  One of the datasets in the test is being created a private and we don't know why or how.  Can you take a look at it and see if you find anything?  Or can we pair at some point?

**Comment by nickumia-reisys** [comment]
Tests are passing.. TODO:
- [ ] Update README
- [ ] Sync `CKAN 2.10` templates
- [ ] Any other random changes... TBD

**Comment by FuhuXia** [comment]
After data is loaded into ckan 210, more issues surfaced. Here are the issues on the landing page.
 
image

**Comment by FuhuXia** [comment]
Looks good and ready to be released as version='0.2.0',

**Comment by FuhuXia** [comment]
manually publish to PYPI  version 0.2.0.

File: ckanext/datagovtheme/public/css/datagovtheme.css [hunk 1]
```diff
@@ -18,6 +18,13 @@
     background-color: transparent;
     border: 0;
 }
+.hide {
```
Comment (btylerburton) [inline ckanext/datagovtheme/public/css/datagovtheme.css:21]: anyone know the reason for this alt CSS file?

File: ckanext/datagovtheme/templates/snippets/sort_by.html [hunk 2]
```diff
@@ -8,15 +8,15 @@
   {% snippet 'snippets/sort_by.html', sort=c.sort_by_selected %}
 
 #}
-<span class="form-select {{ 'form-group' }} control-order-by">
+<div class="form-group control-order-by">
```
Comment (btylerburton) [inline ckanext/datagovtheme/templates/snippets/sort_by.html:11]: what's the template tags for? `{{ 'form-group'}}`

also, looks like we deleted `form-select`??

**Review by btylerburton — COMMENTED** [review]
a few questions...

File: Makefile [hunk 3]
```diff
@@ -1,4 +1,4 @@
-CKAN_VERSION ?= 2.9
+CKAN_VERSION ?= 2.10
```
[the remark this answers is not in the document]
Comment (FuhuXia) [reply Makefile:1]: Just like we did in 2.9 in the line being replace, we did not specify the patch version.

**Review by btylerburton — APPROVED** [review]

File: ckanext/datagovtheme/public/css/datagovtheme.css (hunk 1 above)
Comment (nickumia-reisys) [reply ckanext/datagovtheme/public/css/datagovtheme.css:21]: Apparently CKAN CSS messed with Data.gov theme CSS and USWDS CSS... this is where we use it @btylerburton 
```html
ckanext/datagovtheme/templates/header-site-usage.html:3:<a href="#content" class="hide">{{ _('Skip to content') }}</a>
ckanext/datagovtheme/templates/header-site-usage.html:18:                        <label class="hide" for="search-header">Search for:</label>
ckanext/datagovtheme/templates/package/search.html:15:          <label for="search-big" class="hide">{{ _('Search datasets') }}</label>
ckanext/datagovtheme/templates/package/search.html:124:        <label for="field-location" class="hide">{{ _('Filter by location') }}</label>
ckanext/datagovtheme/templates/header.html:63:<a href="#content" class="hide">{{ _('Skip to content') }}</a>
ckanext/datagovtheme/templates/header.html:104:                            <li class="usa-nav__primary-item desktop-hide">
```

**Pull request merged**